assets/wizards/engagement/components/prompt.tsx

```diff
@@ -0,0 +1,303 @@
+													checked={ values[ field.name ]?.indexOf( option.id ) > -1 }
+													onChange={ ( value: boolean ) => {
+														const toUpdate = { ...values };
+														if ( ! value && toUpdate[ field.name ].indexOf( option.id ) > -1 ) {
+															toUpdate[ field.name ].value = toUpdate[ field.name ].splice(
+																toUpdate[ field.name ].indexOf( option.id ),
+																1
+															);
+															setIsDirty( true );
+														}
+														if ( value && toUpdate[ field.name ].indexOf( option.id ) === -1 ) {
+															toUpdate[ field.name ].push( option.id );
+															setIsDirty( true );
+														}
+														setValues( toUpdate );
+													} }
+												/>
+											</BaseControl>
+										) ) }
+									</BaseControl>
+								) }
+								{ 'string' === field.type && field.max_length && 100 < field.max_length && (
+									<TextareaControl
+										className="newspack-textarea-control"
+										label={ field.label }
+										disabled={ inFlight }
+										help={ `${ values[ field.name ]?.length || 0 } / ${ field.max_length }` }
+										onChange={ ( value: string ) => {
+											if ( value.length > field.max_length ) {
+												return;
+											}
+
+											const toUpdate = { ...values };
+											toUpdate[ field.name ] = value;
+											if ( JSON.stringify( toUpdate ) !== JSON.stringify( values ) ) {
+												setIsDirty( true );
+											}
+											setValues( toUpdate );
+										} }
+										placeholder={ field.default }
+										rows={ 10 }
+										value={ values[ field.name ] || '' }
+									/>
+								) }
+								{ 'string' === field.type && field.max_length && 100 >= field.max_length && (
+									<TextControl
+										label={ field.label }
+										disabled={ inFlight }
+										help={ `${ values[ field.name ]?.length || 0 } / ${ field.max_length }` }
+										onChange={ ( value: string ) => {
+											if ( value.length > field.max_length ) {
+												return;
+											}
+
+											const toUpdate = { ...values };
+											toUpdate[ field.name ] = value;
+											if ( JSON.stringify( toUpdate ) !== JSON.stringify( values ) ) {
+												setIsDirty( true );
+											}
+											setValues( toUpdate );
+										} }
+										placeholder={ field.default }
+										value={ values[ field.name ] || '' }
+									/>
+								) }
+								{ 'int' === field.type && 'featured_image_id' === field.name && (
+									<BaseControl
+										id={ `newspack-engagement-wizard__${ field.name }` }
+										label={ field.label }
+									>
+										<ImageUpload
+											buttonLabel={ __( 'Choose image', 'newspack' ) }
+											disabled={ inFlight }
+											image={ image }
+											onChange={ ( attachment: Attachment ) => {
+												const toUpdate = { ...values };
+												toUpdate[ field.name ] = attachment?.id || 0;
```

Comment:
I think we're missing the `id` property on `Attachment` type

Reply:
Fixed in 681c89c!